nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/dynamodb/GetDynamoDB.java

```diff
@@ -0,0 +1,180 @@
+        if (keysToFlowFileMap.isEmpty()) {
+            return;
+        }
+
+        final DynamoDB dynamoDB = getDynamoDB();
+
+        try {
+            BatchGetItemOutcome result = dynamoDB.batchGetItem(tableKeysAndAttributes);
+
+            // Handle processed items and get the json document
+            List<Item> items = result.getTableItems().get(table);
+            for (Item item : items) {
+                ItemKeys itemKeys = new ItemKeys(item.get(hashKeyName), item.get(rangeKeyName));
+                FlowFile flowFile = keysToFlowFileMap.get(itemKeys);
+
+                ByteArrayInputStream bais = new ByteArrayInputStream(item.getJSON(jsonDocument).getBytes());
+                flowFile = session.importFrom(bais, flowFile);
+
+                session.transfer(flowFile,REL_SUCCESS);
+                keysToFlowFileMap.remove(itemKeys);
+            }
+
+            // Handle unprocessed keys
+            Map<String, KeysAndAttributes> unprocessedKeys = result.getUnprocessedKeys();
+            if ( unprocessedKeys != null && unprocessedKeys.size() > 0) {
+                KeysAndAttributes keysAndAttributes = unprocessedKeys.get(table);
+                List<Map<String, AttributeValue>> keys = keysAndAttributes.getKeys();
+
+                for (Map<String,AttributeValue> unprocessedKey : keys) {
+                    Object hashKeyValue = unprocessedKey.get(hashKeyName);
+                    Object rangeKeyValue = unprocessedKey.get(rangeKeyName);
+                    sendUnhandledToFailure(session, keysToFlowFileMap, hashKeyValue, rangeKeyValue);
+                }
+            }
+
+            // Handle any remaining items
+            for (ItemKeys key : keysToFlowFileMap.keySet()) {
+                FlowFile flowFile = keysToFlowFileMap.get(key);
+                flowFile = session.putAttribute(flowFile, DYNAMODB_KEY_ERROR_NOT_FOUND, DYNAMODB_KEY_ERROR_NOT_FOUND_MESSAGE + key.toString() );
+                session.transfer(flowFile,REL_FAILURE);
+                keysToFlowFileMap.remove(key);
+            }
+
+        } catch(AmazonServiceException exception) {
+            getLogger().error("Could not process flowFiles due to exception : " + exception.getMessage());
+            List<FlowFile> failedFlowFiles = processException(session, flowFiles, exception);
+            session.transfer(failedFlowFiles, REL_FAILURE);
+        } catch(AmazonClientException exception) {
+            getLogger().error("Could not process flowFiles due to exception : " + exception.getMessage());
+            List<FlowFile> failedFlowFiles = processException(session, flowFiles, exception);
+            session.transfer(failedFlowFiles, REL_FAILURE);
+        } catch(Exception exception) {
+            getLogger().error("Could not process flowFiles due to exception : " + exception.getMessage());
+            List<FlowFile> failedFlowFiles = processException(session, flowFiles, exception);
+            session.transfer(failedFlowFiles, REL_FAILURE);
```

Comment:
Same as above - was using method overloading to process specific exception since they have different attributes but have now changed names of process exception method.